components/script_traits/lib.rs

```diff
@@ -121,6 +122,9 @@ pub enum LayoutControlMsg {
 /// parameters or headers
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct LoadData {
+    /// The origin of the page when it started the load.
+    /// Used for same-origin comparisons.
+    pub source_origin: ImmutableOrigin,
```

Comment:
I'm confused, the comment makes it sound like this is the origin of the page that triggered a navigation, but when `LoadData::new` is called by the constellation, it's set to the origin of the URL.

Reply:
Yes currently, it's a bit of a hack, and it's only meant for when the `LoadData` is created from script. The ones created in the constellation are I think either top-level loads initiated by the embedder, or part of the "create a new iframe or auxiliary" flows which I don't think involve any potential JS URL(those would happen in a later once the about:blank iframe or auxiliary has been created and is then navigated).

I think this needs revisiting as @jdm suggested, maybe with a new `NavigationOrigin` concept, that would have things like `TopLevel` and `Script(ImmutableOrigin or String of Origin)` and so on.

Reply:
And by the way the goal is to do those JS url evaluations in a subsequent task queued from the navigation steps, and prevent the security problems noted at https://github.com/whatwg/html/issues/2591

Reply:
See https://github.com/servo/servo/issues/22879

Reply:
OK, this is possibly just bikeshedding the comment and field name.